ui/src/map/layers/highlight/duplicate-split-section-highlight-layer.ts

```diff
@@ -13,42 +13,104 @@ import { LineString } from 'ol/geom';
 import Feature from 'ol/Feature';
 import {
     blueSplitSectionStyle,
+    notOverlappingDuplicateSplitSectionStyle,
     redSplitSectionStyle,
 } from 'map/layers/utils/highlight-layer-utils';
 import { LocationTrackId } from 'track-layout/track-layout-model';
 import { SplittingState } from 'tool-panel/location-track/split-store';
-import { getLocationTrackInfoboxExtras } from 'track-layout/layout-location-track-api';
+import {
+    getLocationTrackInfoboxExtras,
+    SplitDuplicate,
+} from 'track-layout/layout-location-track-api';
 import { filterNotEmpty } from 'utils/array-utils';
 import { LayoutContext } from 'common/common-model';
 
 function createFeatures(
     alignments: LocationTrackAlignmentDataHolder[],
-    duplicateIds: LocationTrackId[],
+    duplicates: SplitDuplicate[],
     linkedDuplicates: LocationTrackId[],
 ): Feature<LineString>[] {
     return alignments
-        .filter((alignment) => duplicateIds.includes(alignment.header.id))
-        .flatMap(({ points, header }) => {
-            const polyline = points.map(pointToCoords);
-            const lineString = new LineString(polyline);
-            const feature = new Feature({ geometry: lineString });
+        .flatMap((alignment) => {
```

Comment:
Tämä on aika pitkä lambda. Voisi harkita että täältä sisältä pilkkoisi juttuja omiksi funktioikseen, jolloin tästä tulisi helpommin luettava. Esimerkiksi tuo `pointCollections`:in luonti ja `features`:in luonnissa käytetty `map`:in lambda voisivat olla hyviä kandidaatteja tällaiselle.